lib/ravelin.rb

```diff
@@ -1,5 +1,39 @@
-require "ravelin/version"
+require 'date'
```

Comment:
As any other thing it comes a lot to developer taste or existing guidelines.

Personally the reason why I dislike this style is it reminds me the Rails require the world mindset and then you pay on anarchy and slowness :). Some of the problems as expected: 
- magically you can reference other classes on the code without knowing when they come from, which might then lead to hard to trace bugs of lookup name constants.
- to run a single test file, it has to require everything first: do i need the whole Faraday library to run a single test? 

I also like this metric of understanding the bloat of any class by knowing how many other classes it depends on. Sometimes it's easier to count require statements than counting constants and refactor on that premises.

The style I like to follow is:
- for a given class file, see on top all the requires for other namespaces/classes it needs - my "definition" of needs is given by only the class constants on the file. If that class then requires another class/namespace, it will be its responsibility to set the require.
- any spec helper test file, starts by requiring spec_helper and then the class/module whatever it is testing.

If you have to require multiple times the same thing that's acceptable. Nowadays is extremely cheap O(1) for Ruby to check if something has been required already. 

You will notice on your code that you are referencing Faraday on the top domain so you are already paying the bloat here for anytime you require the `Javelin` namespace. Some gems resolve this by offering a different class (outside of main namespace) to avoid this issue. 

Consider this more of a food for thought comment. I think the work on this PR is pretty solid :+1:  and I don't want to block you. The above is also a result of a bit of frustrating dealing with other gems in the past. Maybe something for the future.